velox/expression/Expr.cpp

```diff
@@ -901,6 +901,20 @@ void Expr::evalAll(
     }
   }
 
+  if (context->errors()) {
+    if (remainingRows == &rows) {
+      nonNulls.allocate(rows.end());
```

Comment:
Maybe I am missing something, but why allocate when you are going to set it to rows below ?

Reply:
(that was done as part of PR[1081](https://github.com/facebookincubator/velox/pull/1081)

Reply:
(that was done as part of [PR1081](https://github.com/facebookincubator/velox/pull/1081))

nonNulls is a LocalSelectivityVector which has a unique_ptr to a SelectivityVector as one of its fields.  Calling allocate initializes the unique_ptr to point to a SelectivityVector of the given size (instead of being a nullptr), and initializes the bit vector bits_ in the SelectivityVector to the given size.  So in the next line, nonNulls().get() doesn't return a nullptr, and when the copy constructor for the bit vector is called, it can just assigns the uint64_t values in the vector to those in rows.  Since that assignment would call resize anyway, there's no additional cost.  Also, allocate can grab a SelectivityVector out of a pool that the ExecContext maintains so it may be cheaper from a memory allocations perspective.

(I'm not so clever that I figured this out, I just copied what was done in the for loop above when defaultNullBehavior is enabled and one of the arguments is null).